### DISCUSSION TITLE
Problem with attachments while importing bb app

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#9041] [Help] [answered]
I just tried to import a bb app from my bb dev platform, to the bb production platform.
I have a problem with a inner bb table that contains attachments : as you can see, some pictures passed OK, some others respond a 404 error.

I  think that the root cause is related to the improvements you made on apps export/import.

The KO ones are the oldest images.
The OK images are the newest ones, updated AFTER your improvements.

Do you see what could be done, without having to manually delete / import the pictures again ?

Thanks for your help.

<img width="481" alt="image" src="https://user-images.githubusercontent.com/28592069/207543787-e8f3b036-90ab-473c-b2e8-a6114394fa80.png">

**Comment by NicolasFabre81** [comment, marked as answer]
I completely deleted and re-uploaded all my images, but the problem remains : 30% of the attachment rows are OK, 70% are KO.
If I look at the image urls, the OK ones have the good App ID, wheres the KO ones are still on the dev environment app Id : 
OK : https://bb/prod-budi-app-assets/**app_f7657ec140df4ae288ae8b0fe972faa1**/attachments/a07cb2d9-c548-4459-95f5-ca441665b005.JPG
KO : https://bb/prod-budi-app-assets/**app_38fe9b230042408dba465afe8e78a70d**/attachments/46e0ef62-ed38-48fa-a07e-5084a216ac60.JPG
Everything happens as if the import process of attachments stops at a moment, leaving some files unprocessed...
@mielune

**Reply by mike12345567** [reply]
Hi @NicolasFabre81 - can you provide an app export which experiences this problem? Would make debugging a lot easier.

**Reply by NicolasFabre81** [reply]
I'm afraid this is not possible, due to data confidentiality (working for pharma labo)...
I'm gonna try to make a mini-app that shows the problem

**Reply by mike12345567** [reply]
Hi @NicolasFabre81 - yes if you could build up an app that exhibits the behaviour that can be exported that would be perfect!

**Reply by NicolasFabre81** [reply]
Hi @mike12345567 , I think that to -maybe- reproduce it I need a table with a lot of images, not so easy to auto-generate such data in a bb table...

**Reply by NicolasFabre81** [reply]
I don't manage to reproduce it with a brand new app... The import / export runs well, and all photo's url are transformed with the new app ID.
In my problem, some photo's url kepp the old app ID, causing the photo preview to fail, whereas the photo file is present in the attachments folder.

**Reply by NicolasFabre81** [reply]
Hi @mike12345567 , do you know how, or at which moment, the attachment URL is generated while we import an app ?

**Reply by NicolasFabre81** [reply]
Hello @mike12345567 ,
I finally managed to make a little app to show you the problem.
You can download it here : [https://pf.sharedoc.fr/fastlink.php?d=LWZsaW5rOjUwNTUxOmZsaW5rLS1yYW5kMTohNC1tISM4czpyYW5kMS0tcmFuZDI6I2hAa3N6IXI6cmFuZDItLWRlc3Q6ZmFicmVuaUBwaWVycmUtZmFicmUuY29tOmRlc3Qt](url)
While importing it, you will see that some of the photos does not display.
It appears only starting from about 25 photos.

**Reply by mike12345567** [reply]
Hi @NicolasFabre81 - thank you for putting this together - I've been able to replicate your issue from this - I will get a fix out for it.

**Reply by NicolasFabre81** [reply]
Hi @mike12345567 , it would be great ! Do you plan to make it for the next release ?

**Reply by mike12345567** [reply]
Hi @NicolasFabre81 - hopefully will have it out in a patch release, likely some point next week.

**Reply by NicolasFabre81** [reply]
Hello @mike12345567 , I confirm you this is solved in V2.4
